app/views/split_checkout/_payment.html.haml

```diff
@@ -1,5 +1,7 @@
 .medium-6
   %div.checkout-substep{"data-controller": "paymentmethod"}
+    = render partial: "split_checkout/voucher_section", formats: [:cable_ready], locals: { order: @order, voucher_adjustment: @voucher_adjustment }
```

Comment:
This `formats: [:cable_ready]` looks really weird to me and I think we don't actually need it, along with the multiple-format-rendering in the related controllers and the `.cable_ready.haml` extension on the `views/split_checkout/_voucher_section` (it can just be regular `.html.haml`). 

I'll take a look at it...

Reply:
if you have the partial as a regular `.html.haml` then this will break (as you can see in the failing test in your PR) : 
```ruby
  def render_voucher_section
    render(
      status: :ok,
      cable_ready: cable_car.replace(
        "#voucher-section",
        partial(
          "split_checkout/voucher_section",
          locals: { order: @order, voucher_adjustment: @order.voucher_adjustments.first }
        )
      )
    )
  end
```
That's why I am using `formats: [:cable_ready]` . I agree it looks weird but I couldn't find any other solution, do you have any suggestions ?

Reply:
Ah, I missed those failing request specs on the `VoucherAdjustmentsController`. Okay, so this is partly why I was suggesting the changes in that other commit https://github.com/rioug/openfoodnetwork/commit/71e92aec6e231c0260955ba116a01a77b666c6b2, which is that the code path that runs when you actually use the app and the code path that is being encountered in those tests are actually two separate code paths with different outcomes, and it's really tricky to see which is which by just looking at the code in the specs.

Removing the explicit setting of request headers in that spec and removing the last test in it fixes the tests, and there's nothing to fix outside of the specs because it doesn't actually throw that error outside of the scenarios in the test suite.